File: Baekjoon/Class2/27.2839 설탕 배달.py
# 반복 동적 프로그래밍
import sys
input = sys.stdin.readline

INF = 5001

n = int(input())
bag_counts = [INF] * (n + 1) #메모이제이션
bag_counts[3] = 1
if n >= 5: bag_counts[5] = 1

for i in range(6, n + 1):
    bag_counts[i] = min(bag_counts[i - 3], bag_counts[i - 5]) + 1; #더 적은 봉지 수에 하나 추가

if bag_counts[n] >= INF: print("-1")
else: print(bag_counts[n])

''' 그리디
import sys
input = sys.stdin.readline

n = int(input())
bag_count = -1

for five_kg in range(n // 5, -1, -1): #5kg 봉지 최대부터 0개까지
    remain = n - (5 * five_kg)
    if remain % 3 == 0: #나머지가 3으로 나눠 떨어지면 종료
        bag_count = five_kg + (remain // 3)
        break
print(bag_count)
'''

''' 재귀 동적 프로그래밍
import sys
sys.setrecursionlimit(2000)
input = sys.stdin.readline

INF = 5001

def count_bags(kg, bag_counts):
    if kg <= 5: return bag_counts[kg]
    if bag_counts[kg] == INF: #아직 계산 안 됨
        #더 적은 봉지 수에 하나 추가
        bag_counts[kg] = min(count_bags(kg - 3, bag_counts), count_bags(kg - 5, bag_counts)) + 1
    return bag_counts[kg]

def main():
    n = int(input())
    bag_counts = [INF] * (n + 1) #메모이제이션
    bag_counts[3] = 1
    if n >= 5: bag_counts[5] = 1

    bag_count = count_bags(n, bag_counts)
    if bag_count >= INF: print("-1")
    else: print(bag_count)

main()
'''

File: Baekjoon/Class2/27.2839 설탕 배달.js
// 반복 동적 프로그래밍
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const INF = 5001;

const n = Number(input);
const bagCounts = new Array(n + 1).fill(INF); // 타뷸레이션
bagCounts[3] = 1;
if (n >= 5) bagCounts[5] = 1;

for(let i = 6; i <= n; i++)
    bagCounts[i] = Math.min(bagCounts[i - 3], bagCounts[i - 5]) + 1; // 더 적은 봉지 수에 하나 추가

if(bagCounts[n] >= INF) console.log("-1");
else console.log(bagCounts[n]);

/* 그리디
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const n = Number(input);
let bagCount = -1;

for(let fiveKg = Math.floor(n / 5); fiveKg >= 0; fiveKg--) { // 5kg 봉지 최대부터 0개까지
    const remain = n - (5 * fiveKg);
    if(remain % 3 === 0) { // 나머지가 3으로 나눠 떨어지면 종료
        bagCount = fiveKg + Math.floor(remain / 3);
        break;
    }
}
console.log(bagCount);
*/

/* 재귀 동적 프로그래밍
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const INF = 5001;

const countBags = (kg, bagCounts) => {
    if(kg <= 5) return bagCounts[kg];
    if(bagCounts[kg] === INF) // 아직 계산 안 됨
        // 더 적은 봉지 수에 하나 추가
        bagCounts[kg] = Math.min(countBags(kg - 3, bagCounts), countBags(kg - 5, bagCounts)) + 1;
    return bagCounts[kg];
}

const n = Number(input);
const bagCounts = new Array(n + 1).fill(INF); // 메모이제이션
bagCounts[3] = 1;
if (n >= 5) bagCounts[5] = 1;

const bagCount = countBags(n, bagCounts);
if(bagCount >= INF) console.log("-1");
else console.log(bagCount);
*/